sdclientapi/__init__.py

```diff
@@ -445,9 +448,13 @@ def download_submission(
 
         :returns: Tuple of sha256sum and path of the saved submission.
         """
-        path_query = "api/v1/sources/{}/submissions/{}/download".format(
-            submission.source_uuid, submission.uuid
-        )
+
+        if submission.download_url != "":
+            path_query = submission.download_url
+        else:
+            path_query = "api/v1/sources/{}/submissions/{}".format(
+                submission.source_uuid, submission.uuid
```

Comment:
similarly here, we should fix the 404 exception (using URL in exception if uuid is empty) and since we use `submission.filename` below, we should generate a filename (another UUID is not a bad choice imho) for the case where it is empty